juju/juju
# juju add-unit --attach-storage

## Description of change

Update the add-unit command, API, API server and
state code to enable users to add a new unit to
an application, attaching existing storage.

## QA steps

1. juju bootstrap aws
2. juju deploy postgresql --storage pgdata=ebs,10G
(wait for idle)
3. juju remove-unit postgresql/0
(wait till machine 0 is removed; storage should remain)
4. juju add-unit postgresql --attach-storage pgdata/0
(wait for idle)

## Documentation changes

Yes, we will need to document the new flag.

## Bug reference

None.
